crates/wasmtime/build.rs

```diff
@@ -1,27 +1,52 @@
 fn main() {
     println!("cargo:rerun-if-changed=build.rs");
 
+    // NB: duplicating a workaround in the wasmtime-fiber build script.
+    println!("cargo:rustc-check-cfg=cfg(asan)");
+    if cfg_is("sanitize", "address") {
+        println!("cargo:rustc-cfg=asan");
+    }
+
+    let unix = cfg("unix");
+    let windows = cfg("windows");
+    let miri = cfg("miri");
+    let supported_platform = unix || windows;
+
+    let has_native_signals =
+        !miri && (supported_platform || cfg!(feature = "custom-native-signals"));
+    let has_virtual_memory = supported_platform || cfg!(feature = "custom-virtual-memory");
```

Comment:
wasm32-unknown-emscripten is marked as unix, but has neither signals, nor virtual memory. I guess it is not all too important that Wasmtime works on Emscripten though.

Reply:
That's a good point yeah, although this `unix || windows` logic matches [what's currently in the `sys` module](https://github.com/bytecodealliance/wasmtime/blob/51e28c757e7dbeb434afddcf3961c35959408ba9/crates/wasmtime/src/runtime/vm/sys/mod.rs#L49-L63) so we technically already didn't support Emscripten. It should be reasonable though to add support here though in the future.